Add Update & GetAll crud controllers

Nek- opened the pull request:
This commit comes with some improvements:
- Security check is generalilzed
- Tests are improved
- CustomResponseEvent (luckily not used anywhere ATM) is fixed
- GetAll & Update controllers
- New features for dataprovider, which means BC Break, which means **version upgrade needed to 0.2**

Commit:
Add Update crud controller

This commit comes with some improvements:
- Security check is generalilzed
- Tests are improved
- CustomResponseEvent (luckily not used anywhere ATM) is fixed

Commit:
Add GetAll crud controller

This commit also fixes some mistakes in update controller.

The getall controller comes with a management of pagination.

mcsky commented on src/Serialization/Json/OkContentNormalizer.php:
```diff
@@ -39,7 +46,43 @@ public function normalize($object, $format = null, array $context = array())
+
+            $result['links'] = [
+                'prev' => $previousPage,
+                'next' => $nextPage,
```
In case `$content->hasPreviousPage()` or `$content->hasNextPage()` are false these keys shouldn't be add to response. Current request's url isn't revelant

Nek- replied:
Well, api-json say they should be here. So here they are 😇 .

Nek- replied:
Mybad, you're absolutely right !

`Keys *MUST* either be omitted or have a null value to indicate that a particular link is unavailable.`

mcsky commented on tests/Melodiia/Bridge/Doctrine/DoctrineDataStoreTest.php:
```diff
@@ -22,8 +23,10 @@ public function testItSaveUsingTheEntityManager()
-        $datastore = new DoctrineDataStore($entityManager->reveal());
+        $datastore = new DoctrineDataStore($managerRegistry->reveal());
         $datastore->save($obj);
     }
```
Missing test for the `getPaginated` method :)

Nek- replied:
Fixed

mcsky commented on tests/Melodiia/Crud/Controller/GetAllTest.php:
```diff
@@ -0,0 +1,90 @@
+        $this->assertInstanceOf(CrudControllerInterface::class, $this->controller);
+    }
+
+    public function testItReturnResourceFromDataStoreInsideOkContent()
```
`testItSetResourceFromDataStoreInsideOkContent`

Nek- replied:
I like how I say it, if you don't mind !

mcsky reviewed (changes requested):
Nice work 👍

Commit:
Review PR #13 : Add missing test & jsonapi fix

Commit:
Fix update issue

Problem: the update controller was not updating properly data.
Also the ApiForm was not really prepared for usage of object with
constructors.

This commit fixes everything and externalize the work of datamapping to
an actual data-mapper.

mcsky commented on src/Crud/Controller/Create.php:
```diff
@@ -34,22 +36,32 @@ final class Create implements CrudControllerInterface
         // Metadata you can specify in routing definition
         $modelClass = $request->attributes->get(self::MODEL_ATTRIBUTE);
         $form = $request->attributes->get(self::FORM_ATTRIBUTE);
+        $securityCheck = $request->attributes->get(self::SECURITY_CHECK, null);
```
![image](https://user-images.githubusercontent.com/14903820/54519117-0e7bd880-4966-11e9-92c1-78376f93e3f9.png)

It is normal? I don't figure out where do you add your attributes to the request. 

Anyway, this parameter shouldn't be sent by the client. It could be hardcoded in configuration or retrieve by a custom service (if user have to perform some domain logic)

Nek- replied:
It's only used when you use it. Documentation is definitely missing. It will be my next move !
